sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/RequireFailForTryCatchInJunitCheck.java

```diff
@@ -0,0 +1,215 @@
+ * If the try completes normally the test will pass successfully and skip over any assertions in
+ * the catch block.
+ * If tests are not expecting exceptions, then they should remove the catch block and propagate
+ * the exception to the junit caller which will display the full exception to the user.
+ * </p>
+ * <p>
+ * An example of how to configure the check is:
+ * </p>
+ * <pre>
+ * &lt;module name="RequireFailForTryCatchInJunitCheck"/&gt;
+ * </pre>
+ * @author Richard Veach
+ */
+public class RequireFailForTryCatchInJunitCheck extends AbstractCheck {
+    /**
+     * Violation message key.
+     */
+    public static final String MSG_KEY = "require.fail";
+
+    /**
+     * Fully qualified junit test annotation.
+     */
+    private static final String FQ_JUNIT_TEST = "org.junit.Test";
+    /**
+     * JUnit's fail assertion method name.
+     */
+    private static final String FAIL = "fail";
+
+    /**
+     * {@code true} if the junit test is imported.
+     */
+    private boolean importTest;
+    /**
+     * {@code true} if the junit assert is imported.
+     */
+    private boolean importAssert;
+    /**
+     * {@code true} if the junit fail assertion method is statically imported.
+     */
+    private boolean importStaticFail;
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {
+            TokenTypes.IMPORT,
+            TokenTypes.STATIC_IMPORT,
+            TokenTypes.LITERAL_TRY,
+        };
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public void beginTree(DetailAST rootAST) {
+        importTest = false;
+        importAssert = false;
+        importStaticFail = false;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        switch (ast.getType()) {
+            case TokenTypes.IMPORT:
+                final String imprt = getImportText(ast);
+
+                if (FQ_JUNIT_TEST.equals(imprt)) {
+                    importTest = true;
+                }
+                if ("org.junit.Assert".equals(imprt)) {
+                    importAssert = true;
+                }
+                break;
+            case TokenTypes.STATIC_IMPORT:
+                final String staticImprt = getImportText(ast);
+
+                if ("org.junit.Assert.fail".equals(staticImprt)) {
+                    importStaticFail = true;
+                }
+                break;
+            case TokenTypes.LITERAL_TRY:
+                examineTry(ast);
+                break;
+            default:
+                Utils.reportInvalidToken(ast.getType());
+                break;
+        }
+    }
+
+    /**
+     * Examines the try block for violations.
+     * @param ast The try block to examine.
+     */
+    private void examineTry(DetailAST ast) {
+        final DetailAST method = getStopParent(ast);
```

Comment:
> getStopParent

hard to understand....it is better to name it `getMethod()` as you really care only about methods. In case of lambda this method should return null.

Reply:
@romani I originally had it as `getMethod` but then I thought the addition of lambdas broke that as the name doesn't warn about lambdas. I couldn't think of a good name, hence the stop parent.

I will rename this to `getMethod` again.

Reply:
Done.